src/meta/src/barrier/progress.rs

```diff
@@ -483,7 +492,7 @@ impl CreateMviewProgressTracker {
             upstream_total_key_count,
             definition,
         );
-        if *ddl_type == DdlType::Sink {
+        if *ddl_type == DdlType::Sink && *create_type == CreateType::Background {
             // We return the original tracking job immediately.
             // This is because sink can be decoupled with backfill progress.
             // We don't need to wait for sink to finish backfill.
```

Comment:
Is this path still necessary? Can we now simply treat sinks like materialized views?

Reply:
I suppose so. But it will require more refactoring of code.

Sink is unique from materialized view in that the output will immediately be visible, since it's external to the stream job.
For materialized views, we have to do something extra, which is to hide the output until backfill is complete.

As such we persist materialized views twice, when they undergo background ddl:
1. On initial creation. (mark as `Creating`, i.e. invisible, but recoverable).
2. After creation succeeds (mark as `Finished`, i.e. visible).

For sink we can immediately mark it as `Finished`.

For Index, we have to also go through the same flow as materialized view. When I support `Index` I will also review to see if Sink can be supported likewise and unify more branches of code.

For now I think this is acceptable approach which requires least changes, and with simple concept.

Reply:
> For sink we can immediately mark it as `Finished` (i.e. just step 2), we can't do that for MVs.

This is interesting to realize. 😄